### ISSUE TITLE
Multiplexer and dev cli issues

### ISSUE THREAD

**Issue by alekseyrozh** [sst/sst#4189]
sst v3.0.83

1) Cannot cmd+click on links in multiplexer - already mentioned in https://github.com/sst/ion/issues/827
2) Cannot copy multiline strings in multiplexer - when I highlight multiple lines in multiplexer and copy, it copied only one of them
3) Scroll via 2 finger trackpad gesture on a mac is way too fast which makes it impossible to scroll the same way you do with all other terminals
4) NextJs app gets restarted from time to time resulting in a mess of processes launched on different ports:
<img width="627" alt="Screenshot 2024-09-02 at 12 44 35" src="https://github.com/user-attachments/assets/085a433b-4ed5-47c3-a736-af1ddfcd7cab">

5) --mode=basic often just hangs and doesn't start anything, while multiplexer works just fine
[sst.log](https://github.com/user-attachments/files/16840762/sst.log)

**Comment by gabrielcardetti** [comment]
I have similar issue with this, with a new sst ion proyect, I'm using zsh and vscode terminal and when I do a select in the console a enter to copy the text doesn't copy on the clipboard also I can not click on the links.

I tried to use the --mode=basic but it doesn't works neither
when i run 
````
sst dev --mode=basic --print-logs
```
It hangs for ever on 
time=2024-10-25T00:06:46.678+02:00 level=INFO msg="mqtt connected"
time=2024-10-25T00:06:47.195+02:00 level=INFO msg="connected to iot"

If i just do sst dev it deploys the lambda fine but i can't copy or click the terminal

**Comment by alekseyrozh** [comment]
hi @thdxr sadly still struggling with developer experience due to the problems highlighted at the start. My team and the others in this thread would appreciate if this would be looked at, thank you. (sst 3.3.27)

Number 1 issue is the scroll speed on mac touchpad, it's makes it incredibly hard to use. If just this would be resolved, this would be a massive improvement.

I don't know if anything has changed, but now links are clicking fine. The search isn't quite working though, i've attached the video to illustrate the scrolling speed issue and the search in a VScode terminal.

https://github.com/user-attachments/assets/f5e78815-7389-436f-9e59-a9fd3c4bb0a3

**Comment by jayair** [comment]
@alekseyrozh Can you explain the scroll speed thing a bit? I can't tell from the video but is it scrolling too fast?

**Comment by alekseyrozh** [comment]
@jayair that's right, please see a comparison video attached with scrolling in sst multiplexer and outside of it




https://github.com/user-attachments/assets/bf7d00d3-9468-4bbb-a458-904027d32976

**Comment by jayair** [comment]
@thdxr

**Comment by lefnire** [comment]
I can't copy text in dev-mode (multiplexer) in any way. 
* ctrl+c kills; ctrl+shift+c nothing
* right-click nothing
* **enter does nothing**. 

This the default Terminal installed in Ubuntu, fresh install. sst 3.7.15. 

`--mode=basic` hangs. End of `.sst/log/sst.log`:

```
time=2025-02-12T14:23:36.586-07:00 level=INFO msg="found appsync" rest=xxx.appsync-api.us-east-1.amazonaws.com realtime=xxx.appsync-realtime-api.us-east-1.amazonaws.com
time=2025-02-12T14:23:36.586-07:00 level=INFO msg="checking if appsync is ready"
time=2025-02-12T14:23:36.823-07:00 level=INFO msg="INFO connecting service=appsync.connection"
time=2025-02-12T14:23:37.244-07:00 level=INFO msg="INFO connect message service=appsync.connection msg=\"map[connectionTimeoutMs:300000 type:connection_ack]\""
time=2025-02-12T14:23:37.244-07:00 level=INFO msg="subscribing to" prefix=/sst/dispatch/lefnire/in
time=2025-02-12T14:23:37.244-07:00 level=INFO msg="INFO msg service=appsync.connection type=ka id=<nil>"
time=2025-02-12T14:23:37.535-07:00 level=INFO msg="INFO msg service=appsync.connection type=subscribe_success id=xxx"
time=2025-02-12T14:23:37.535-07:00 level=INFO msg="INFO subscribed service=appsync.connection channel=/sst/dispatch/lefnire/in id=xxx"
time=2025-02-12T14:23:37.535-07:00 level=INFO msg=starting service=aws.task
time=2025-02-12T14:24:37.151-07:00 level=INFO msg="INFO msg service=appsync.connection type=ka id=<nil>"
time=2025-02-12T14:25:37.467-07:00 level=INFO msg="INFO msg service=appsync.connection type=ka id=<nil>"
time=2025-02-12T14:26:37.147-07:00 level=INFO msg="INFO msg service=appsync.connection type=ka id=<nil>"
time=2025-02-12T14:27:37.193-07:00 level=INFO msg="INFO msg service=appsync.connection type=ka id=<nil>"
time=2025-02-12T14:28:37.437-07:00 level=INFO msg="INFO msg service=appsync.connection type=ka id=<nil>"
time=2025-02-12T14:28:41.395-07:00 level=INFO msg=interrupted
time=2025-02-12T14:28:41.395-07:00 level=INFO msg=done service=deployer
time=2025-02-12T14:28:41.395-07:00 level=INFO msg=done service=aws.task
time=2025-02-12T14:28:41.395-07:00 level=INFO msg=done service=dev
time=2025-02-12T14:28:41.395-07:00 level=INFO msg=done service=watcher
time=2025-02-12T14:28:41.395-07:00 level=INFO msg="shutting down server" service=server
time=2025-02-12T14:28:41.395-07:00 level=INFO msg="INFO connection closed service=appsync.connection"
time=2025-02-12T14:28:41.395-07:00 level=INFO msg="server done" service=server
time=2025-02-12T14:28:41.395-07:00 level=INFO msg="done mosaic" err="Get \"http://localhost:13557/stream\": dial tcp 127.0.0.1:13557: connect: connection refused"
time=2025-02-12T14:28:41.395-07:00 level=ERROR msg="exited with error" err="Get \"http://localhost:13557/stream\": dial tcp 127.0.0.1:13557: connect: connection refused"

```

Can't do it even via Jetbrains or VS Code, where I'd hoped the IDE had its own sort of ctrl+c / ctrl+shift+c override. Reverted to taking screenshots and having ChatGPT transcribe them for me.

**Comment by lefnire** [comment]
@thdxr I notice you ask what terminal we use, is there a terminal compatible with most operating systems (or just Ubuntu, for my purposes) you know to work? I'm just using the stock Terminal app; maybe there's a more SST-friendly one?

**Comment by phuskey-compassion** [comment]
My team is running a monorepo using Nx. I get the multiplexer issues when trying to run dev mode through Nx. If I use the standalone sst cli `cd apps/{app_name} && sst dev` or run it through pnpm `pnpm --filter {app_name} exec sst dev`, the mulitplexer works fine. Basic mode just hangs no matter which option I try.

Some other things to note. This only seems to happen on our Nextjs apps. We have some Api Gateway apps where the multiplexer works fine with all the above options. Also, I'm running sst on WSL Ubuntu. My coworkers that have Macs don't seem to have these problems at all.
